client.go

```diff
@@ -1026,6 +1030,8 @@ func (c *Client) connect() error {
 		return common.NewInitContextError("connecting to a redis server in the init context is not supported")
 	}
 
+	// If the redisClient is already instantiated, it is safe
+	// to assume that the connection is already established.
```

Comment:
Well we also can have the connection disconnected :shrug: But I would guess that will be rare and we can create an issue and fix it when we start seeing it.